api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -216,6 +218,71 @@ public Patient getPatient(Integer patientId) throws APIException {
 		return dao.getPatient(patientId);
 	}
 	
+	/**
+	 * @see org.openmrs.api.PatientService#getPatients(String, String, List, boolean, String, Integer, Integer, Date)
+	 */
+	
+	@Override
+	public List<Patient> getPatients(String name, String identifier, List<PatientIdentifierType> identifierTypes,
```

Comment:
I think this method should be break in to several other methods: With all these if and else conditions its really hard to read whats going on.

Reply:
I divide this method into two methods, now it becomes easy to understand